package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,383 @@
+        self._setup_frames(traj, start, stop, step)
+
+    def _prepare(self):
+        logger.info('numframes {0}'.format(self.nframes))
+        self.dist_matrix = np.zeros((self.nframes, self.nframes))
+
+    def _single_frame(self):
+        traj_index = self._ts.frame
+        i_ref = self.atoms.positions - self.atoms.center_of_mass()
+        # diagonal entries need not be calculated due to metric(x,x) == 0 in
+        # theory, _ts not updated properly. Possible savings by setting a
+        # cutoff for significant decimal places to sparsify matrix
+        for j in range(self.stop-1, self._ts.frame-1, -self.step):
+            self._ts = self._u.trajectory[j]
+            j_ref = self.atoms.positions-self.atoms.center_of_mass()
+            dist = self._metric(i_ref, j_ref, weights=self._weights)
+            # distance squared in preparation for kernel calculation
+            # don't think this will come up in other areas, but might be
+            # a fix we should make later
+            self.dist_matrix[traj_index, j] = dist**2 if dist > self._cutoff else 0
+            self.dist_matrix[j, traj_index] = self.dist_matrix[traj_index, j]
+
+    def save(self, filename):
+        np.savetxt(filename, self.dist_matrix)
+        logger.info("Wrote the distance-squared matrix to file %r", filename)
+
+
+class Epsilon(object):
+    """Manipulates a distance matrix by local scale parameters
+
+    Attributes
+    ----------
+    scaledMatrix : DistanceMatrix object
+        A matrix with each term divided by a local scale parameter
+
+    Methods
+    -------
+    determineEpsilon()
+        Determine local scale parameters using a chosen algorithm
+    """
+    def __init__(self, DistanceMatrix, **kwargs):
+        self._DistanceMatrix = DistMatrix
+
+    def determine_epsilon(self):
+        pass
+
+
+class EpsilonConstant(Epsilon):
+    """Premade function for the determination of epsilon based on providing
+    an arbitrary constant"""
+
+    def __init__(self, DistanceMatrix, epsilon):
+        """
+        Parameters
+        ----------
+        epsilon : int
+            The value of epsilon to be used as a local scale parameter
+        """
+        self.scaledMatrix = DistanceMatrix.dist_matrix
+        self._epsilon = epsilon
+
+    def determine_epsilon(self):
+        self.scaledMatrix /= self._epsilon
+        return
+
+
+class DiffusionMap(AnalysisBase):
+    """Non-linear dimension reduction method
+
+    Dimension reduction with diffusion mapping of selected structures in a
+    trajectory.
+
+    Attributes
+    ----------
+    eigenvalues: array
+        Eigenvalues of the diffusion map
+    eigenvectors: array
+        Eigenvectors of the diffusion map
+    embedding : array
+        After calling `embedding(num_eigenvectors)` the diffusion map embedding
+        into the lower dimensional diffusion space will exist here.
+
+    Methods
+    -------
+    decompose_kernel()
```

Comment:
to abstract as a name. This requires a user to have a very deep understanding of math and the diffusion map algorithm. We have people using MDAnalysis with a weak math background as well (like chemists and biologists) and we should take care that they aren't intimidated by using badly chosen words.

scikit-learn uses as a standard `fit` to mean the algorithm is calculating it's values on the provided data. This is easy to understand for a lot of people and a general word that we can use for PCA as well and other algorithms.